### PR TITLE
Create token and key flow 

### PR DESCRIPTION
**Pull request by sans-harness**
- Implemented new functionality for creating and managing personal access tokens, including a token creation form and success dialog.
- Added support for SSH key management, with a new form for adding SSH keys and associated dialogs.
- Updated the profile settings page to include buttons for adding new tokens and SSH keys.
- Integrated new components into the Playground package, including TokenCreateDialog, SshKeyCreateDialog, and their respective forms, to provide reusable UI elements for token and SSH key management.
- Demoed in standup on 10/9. 

Complete Flow:

https://github.com/user-attachments/assets/7d7808ae-bcbf-4271-a2a6-dd463f7eb08d

Error handling:

https://github.com/user-attachments/assets/668b0500-8f54-491a-8cc0-d7f61fca6f39

### CODE REVIEW

File: apps/gitness/src/pages/profile-settings/profile-settings-keys-container.tsx [hunk 1]
```diff
@@ -3,17 +3,54 @@ import { useState, useEffect } from 'react'
...
+  const [saveSshKeyDialog, setSshKeyDialog] = useState(false)
+  const [apiError, setApiError] = useState<{
+    type: 'keyFetch' | 'tokenFetch' | 'keyCreate' | 'tokenCreate'
+    message: string
+  } | null>(null)
+
+  const [createdTokenData, setCreatedTokenData] = useState<{
+    identifier: string
+    lifetime: string
+    token: string
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/profile-settings/profile-settings-keys-container.tsx:38]: Can we use existing type here? https://github.com/harness/react-api-client/blob/main/packages/code-service/src/services/schemas/TypesTokenResponse.ts#L6

File: apps/gitness/src/pages/profile-settings/profile-settings-keys-container.tsx [hunk 2]
```diff
@@ -25,22 +62,121 @@ export const SettingsProfileKeysPage = () => {
...
+    }
+  )
+
+  const handleCreateToken = (tokenData: { identifier: string; lifetime: string }) => {
+    const body: CreateTokenRequestBody = {
+      identifier: tokenData.identifier
+    }
+
+    if (tokenData.lifetime.toLowerCase() !== 'never') {
+      const convertedLifetime = parseInt(tokenData.lifetime, 10) * 24 * 60 * 60 * 1000 * 1000000
```
Comment (vardanbansal-harness) [inline apps/gitness/src/pages/profile-settings/profile-settings-keys-container.tsx:135]: nit: Add `24 * 60 * 60 * 1000 * 1000000` as a named constant?

File: packages/canary/src/components/button.tsx [hunk 3]
```diff
@@ -5,7 +5,7 @@ import { cn } from '@/lib/utils'
 import { CanaryOutletFactory, CanaryOutletName } from '@/lib/CanaryOutletFactory'
 
 const buttonVariants = cva(
-  'inline-flex items-center justify-center whitespace-nowrap rounded-[4px] text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50',
+  'inline-flex items-center justify-center whitespace-nowrap rounded-[4px] text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:opacity-50 disabled:cursor-not-allowed',
```
Comment (vardanbansal-harness) [inline packages/canary/src/components/button.tsx:8]: Does this affect all the buttons?

File: packages/playground/src/components/ssh-key-create-form.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                  </Button>
+                  <Button type="submit" size="sm" disabled={!isValid || isSubmitting}>
+                    {!isSubmitting ? 'Save' : 'Saving...'}
+                  </Button>
+                </>
+              ) : (
+                <Button variant="ghost" type="button" size="sm" theme="success" className="pointer-events-none">
+                  Key Saved&nbsp;&nbsp;
+                  <Icon name="tick" size={14} />
+                </Button>
```
Comment (vardanbansal-harness) [inline packages/playground/src/components/ssh-key-create-form.tsx:95]: nit: Probably would be better to put this text and icon in a `flex` and then put `gap` between them?

File: packages/playground/src/components/token-create-form.tsx [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+import { FormFieldSet } from '../index'
+
+const formSchema = z.object({
+  identifier: z.string().min(1, { message: 'Please provide a name' }),
+  lifetime: z.string().min(1, { message: 'Please select an expiration' })
+})
+
+export type TokenFormType = z.infer<typeof formSchema>
+
+const expirationOptions = [
```
Comment (vardanbansal-harness) [inline packages/playground/src/components/token-create-form.tsx:26]: I see this duplicated, lets dedupe?

File: apps/gitness/src/pages/profile-settings/profile-settings-keys-container.tsx (hunk 1 above)
Comment (sans-harness) [reply apps/gitness/src/pages/profile-settings/profile-settings-keys-container.tsx:38]: We could, but it doesn't conform. We cherry pick fields we need for the UI. They're pretty different, not a good idea in my opinion.

```
  const tokenData = {
          identifier: newToken.token?.identifier ?? 'Unknown',
          lifetime: newToken.token?.expires_at
            ? new Date(newToken.token.expires_at).toLocaleDateString()
            : 'No Expiration',
          token: newToken.access_token ?? 'Token not available'
        }
```

File: packages/playground/src/components/ssh-key-create-form.tsx (hunk 4 above)
Comment (sans-harness) [reply packages/playground/src/components/ssh-key-create-form.tsx:95]: key


Not sure if it needs a gap?

Should ideally be conformant across all forms-
https://github.com/harness/canary/blob/main/packages/playground/src/pages/sandbox-repo-create-page.tsx#L240

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**